### PR TITLE
fix admin v1 edit venue functionality

### PR DESCRIPTION
**Pull request by denisdimitrov**
When editing a venue with `parentId` the function updates the sovereign venue with the values from the current one, because they were provided with spread operator. Now the sovereign venue data is being passed instead.

### CODE REVIEW

File: functions/venue.js [hunk 1]
```diff
@@ -636,17 +636,17 @@ exports.updateVenue = functions.https.onCall(async (data, context) => {
...
+  //   !data.placement.state ||
+  //   data.placement.state === PlacementState.SelfPlaced
+  // ) {
+  //   updated.placement = {
+  //     ...data.placement,
+  //     state: PlacementState.SelfPlaced,
+  //   };
+  // } else if (data.placementRequests) {
+  //   updated.placementRequests = data.placementRequests;
+  // }
```
Comment (mike-lvov) [inline functions/venue.js:649]: Why was this commented out?

File: src/pages/Admin/Venue/DetailsForm.tsx [hunk 2]
```diff
@@ -183,14 +183,24 @@ export const DetailsForm: React.FC<DetailsFormProps> = ({
...
+                subtitle:
+                  sovereignVenue.config?.landingPageConfig.subtitle ?? "",
+                description:
+                  sovereignVenue.config?.landingPageConfig.description ?? "",
+                adultContent: sovereignVenue.adultContent ?? false,
+                profile_questions: sovereignVenue.profile_questions,
+                code_of_conduct_questions:
+                  sovereignVenue.code_of_conduct_questions,
                 id: sovereignVenueId,
                 parentId: sovereignVenue?.parentId,
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/DetailsForm.tsx:200]: sovereignVenue can't have a parentId

**Comment by denisdimitrov** [comment]
@dmitrii-zolotukhin Please check this PR for the fix I provided and if you think there is something else that needs to be added feel free to give your input.

File: src/pages/Admin/Venue/DetailsForm.tsx [hunk 3]
```diff
@@ -183,14 +183,24 @@ export const DetailsForm: React.FC<DetailsFormProps> = ({
...
-                ...(vals as VenueInput),
+                name: sovereignVenue.name,
+                subtitle:
+                  sovereignVenue.config?.landingPageConfig.subtitle ?? "",
+                description:
+                  sovereignVenue.config?.landingPageConfig.description ?? "",
+                adultContent: sovereignVenue.adultContent ?? false,
+                profile_questions: sovereignVenue.profile_questions,
+                code_of_conduct_questions:
+                  sovereignVenue.code_of_conduct_questions,
```
Comment (mike-lvov) [inline src/pages/Admin/Venue/DetailsForm.tsx:198]: Why does it update sovereignVenue, not the venue?

**Review by mike-lvov — COMMENTED** [review]
A few questions

File: src/pages/Admin/Venue/DetailsForm.tsx (hunk 3 above)
Comment (denisdimitrov) [reply src/pages/Admin/Venue/DetailsForm.tsx:198]: User statuses are stored in the sovereignVenue that way the whole world can share them. This is why after I update the current venue I need to update the sovereignVenue with the userStatuses only.

File: functions/venue.js (hunk 1 above)
Comment (denisdimitrov) [reply functions/venue.js:649]: This is playa legacy code and it blows up for some of the venues that were created from the admin v3

File: src/pages/Admin/Venue/DetailsForm.tsx (hunk 2 above)
Comment (denisdimitrov) [reply src/pages/Admin/Venue/DetailsForm.tsx:200]: Good point. I will change it.

File: src/pages/Admin/Venue/DetailsForm.tsx (hunk 3 above)
Comment (mike-lvov) [reply src/pages/Admin/Venue/DetailsForm.tsx:198]: Just to be clear. What if you don't pass the field here? Will it be erased?

Comment (mike-lvov) [reply src/pages/Admin/Venue/DetailsForm.tsx:198]: checked the logic, shouldn't do it. Only for the specific fields such as this
image

Comment (mike-lvov) [reply src/pages/Admin/Venue/DetailsForm.tsx:198]: checked the logic, shouldn't do it. Only for the specific fields such as this
image

**Review by mike-lvov — APPROVED** [review]
Looks a bit hacky, but will work

**Comment by denisdimitrov** [comment]
@mike-lvov I just removed the `parentId` from the sovereignVenue now it should work properly.

File: src/pages/Admin/Venue/DetailsForm.tsx (hunk 3 above)
Comment (denisdimitrov) [reply src/pages/Admin/Venue/DetailsForm.tsx:199]: @mike-lvov It's the types that actually force me to pass these fields, I can make them optional. If that's okey.

File: functions/venue.js (hunk 1 above)
Comment (0xdevalias) [reply functions/venue.js:649]: If we haven't already can we please `@debt` it to explain that.

Comment (denisdimitrov) [reply functions/venue.js:649]: @0xdevalias There is debt about it, explaining that this is playa legacy code. We should remove these in the near future.

**Pull request merged**